### PR TITLE
[SPARK-12673][UI] Add missing uri prepending for job description

### PR DESCRIPTION
**Pull request by jerryshao**
Otherwise the url will be failed to proxy to the right one if in YARN mode. Here is the screenshot:

screen shot 2016-01-06 at 5 28 26 pm

### CODE REVIEW

**Comment by jerryshao** [comment]
Jenkins, retest this please.

**Comment by JoshRosen** [comment]
LGTM.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 1]
```diff
@@ -224,7 +224,8 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
       }
       val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
       val formattedSubmissionTime = job.submissionTime.map(UIUtils.formatDate).getOrElse("Unknown")
-      val jobDescription = UIUtils.makeDescription(lastStageDescription, parent.basePath)
+      val basePathUri = UIUtils.prependBaseUri(parent.basePath)
+      val jobDescription = UIUtils.makeDescription(lastStageDescription, basePathUri)
 
       val detailUrl =
         "%s/jobs/job?id=%s".format(UIUtils.prependBaseUri(parent.basePath), job.jobId)
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala]: If you wanted to clean up the code slightly, you could change this to use `basePathUri` defined above, then wrap it back to a single line.

**Comment by zsxwing** [comment]
Ah, didn't notice this before merging #10617. Actually, I prefer this patch since #10617 doesn't add basePath to the url. I'm going to revert it.

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala (hunk 1 above)
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala]: +1

Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala]: I see, I will change it.

**Comment by jerryshao** [comment]
Ahh, looks like #10617 happens to fix the same issue as this PR, I haven't noticed that one...

**Comment by zsxwing** [comment]
LGTM. Merging to master, 1.6 and 1.5

**Pull request closed without merging**